src/Analysis/Ast/Impl/Analyzer/AnalysisModuleKey.cs

```diff
@@ -28,15 +28,14 @@ namespace Microsoft.Python.Analysis.Analyzer {
         public bool IsTypeshed { get; }
         public bool IsNonUserAsDocument { get; }
 
-        public AnalysisModuleKey(IPythonModule module) {
-            Name = module.Name;
-            FilePath = module.ModuleType == ModuleType.CompiledBuiltin ? null : module.FilePath;
-            IsTypeshed = module is StubPythonModule stub && stub.IsTypeshed;
-            IsNonUserAsDocument = (module.IsNonUserFile() || module.IsCompiled()) && module is IDocument document && document.IsOpen;
-        }
+        public AnalysisModuleKey(IPythonModule module) : this(
+            module.Name,
+            module.ModuleType == ModuleType.CompiledBuiltin ? null : module.FilePath,
+            module is StubPythonModule && ((StubPythonModule)module).IsTypeshed,
```

Comment:
`IsTypeshed = module is StubPythonModule stub && stub.IsTypeshed;` is better IMO

Reply:
it can't be used in this( ... ) context in language version it is using. looks like it is supported in never language version if people are okay to move to latest lang version I can make that change.

Reply:
I changed ctor so it can share initialization otherwise when a new field is added, one needs to fix both ctor.

Reply:
We are on .NET Core 3.0 which means C# 8

Reply:
but project is explicitly set to use langversion 7.2

```
    <TargetFramework>netstandard2.0</TargetFramework>
    <RootNamespace>Microsoft.Python.Analysis</RootNamespace>
    <AssemblyName>Microsoft.Python.Analysis</AssemblyName>
  </PropertyGroup>
  <PropertyGroup>
    <!--
      1701, 1702 - "You may need to supply assembly policy"
    -->
    <NoWarn>1701;1702;$(NoWarn)</NoWarn>
    <LangVersion>7.2</LangVersion>
```

Reply:
I can move to 8.0 or latest if team prefer that?

Reply:
Can we do both .NET standard 2.0 and language version 8?

Reply:
I believe we can. Roslyn already do C# 8.0 but still target .net standard 2.0

Reply:
?

Reply:
I mean we are using SDK 3.0 and that automatically mean C# 8

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version

Reply:
We can remove all `LangVersion` now

Reply:
I will send out new PR on removing all lang version